Update Scala.sublime-syntax

jpz opened the pull request:
This small change fixes a bug with the display of the comment `/**/` which was functioning as an unclosed block-comment.

This is fixed by moving the matching of `/**/` ahead of `/**`

jpz commented:
Before and after:

![image](https://cloud.githubusercontent.com/assets/934340/19975336/01ec46be-a205-11e6-946a-a154c0fc1e59.png)

![image](https://cloud.githubusercontent.com/assets/934340/19975355/0ea10ed0-a205-11e6-9af6-460199141b70.png)

wbond commented:
We’ll definitely want to add a test for this.

djspiewak commented:
He beat me too it. :-D Adding a test that some identifier on the line after is *not* a comment should turn the trick.

FichteFoll commented:
Can also target the following newline character.

jpz commented:
Thanks for the reviews - I've added a test.  

I added a trailing value to the test (a simple hex value) to demonstrate that the remainder of the line is normal code.

![image](https://cloud.githubusercontent.com/assets/934340/19996389/a15f2568-a278-11e6-893e-8ecdbfac3d5c.png)

FichteFoll commented on Scala/syntax_test_scala.scala:
```diff
@@ -306,6 +306,10 @@ type Foo = Bar[A] forSome { type A }
 
+  /**/0xff
+//^^^^ comment.block.empty.scala
+//    ^^^^ - thereafter, not a comment 
```
Change this line to

```
//    ^^^^ - comment 
```

jpz replied:
It demonstrates the comment has terminated and text thereafter is not regarded as a comment.  How does describing the hexadecimal number as a comment help in this case?

Kind regards

keith-hall replied:
@jpz it sounds like there is some confusion about how the tests and scope selectors work. If the line starts with a comment, and contains `^` or `<-` then it is a test assertion line (described at http://www.sublimetext.com/docs/3/syntax.html#testing). The rest of the line then dictates the scope selector. Because the scope selector is `- comment` and not just `comment`, it means, assert that the scope does *not* match `comment`. I've never really found any good documentation on scope selectors, but an answer I wrote on SO may be useful: http://stackoverflow.com/questions/37406338/sublime-text-modifying-tmtheme-file/37407588#37407588

jpz replied:
Thanks for explaining and pointing me in the direction of the documentation I should have read :)

Will amend the PR tomorrow.

jpz replied:
done

jpz commented:
Ready to merge @wbond I think

wbond commented:
Thanks for the patch @jpz and the reviews @FichteFoll and @djspiewak!
